lib/ultrajsondec.c

```diff
@@ -131,15 +131,17 @@ static FASTCALL_ATTR JSOBJ FASTCALL_MSVC decode_numeric (struct DecoderState *ds
       case '8':
       case '9':
       {
+        // detect overflow on digit shift
+        if ((intValue * 10ULL) % 10 != 0)
```

Comment:
Would `if (intValue > ULLONG_MAX / 10)` not be equivalent? It's less headache inducing and probably fractionally more performant since the `ULLONG_MAX / 10` will be evaluated at compile time leaving just the `>` operation at runtime.

Reply:
Yeah, I think that's functionally identical. In both cases, an `intValue` of `ULLONG_MAX / 10 = 1844674407370955161` would pass the check, as it should since it does not overflow on the multiplication. It might still overflow on the addition if the next digit is 6 to 9, but that would be caught in the check below. And `intValue = ULLONG_MAX / 10 + 1` would fail this added check in both forms.